src/parseTools.js

```diff
@@ -1292,7 +1292,7 @@ function makeGetTempRet0() {
 }
 
 function makeSetTempRet0(value) {
-  return RELOCATABLE ? "setTempRet0((" + value + ") | 0)" : ("tempRet0 = " + value);
+  return 'asm["setTempRet0"](' + value + ')';
```

Comment:
Sorry, I missed your previous comment.

Relocatable is a mode kind of like dynamic linking. In it the asm.js backend basically replaces each read of a global from a constant address (normal mode) to `global_base` plus a constant offset. Then we can set the global base at runtime, in effect relocating it. We do the same for function pointers too. This allows a simple (but not very fast) form of dynamic linking in asm.js.

I believe this particular line of code needs to check if we are relocatable or not because an additional thing in relocatable mode, which is to not rely on asm.js globals - since we may link multiple relocatable asm.js modules. So we move e.g. `tempRet0` (a global we use to return values larger than JS can fit in a number) from the asm.js module into the outside, where we can only access it with a function call (and then multiple modules can access it). That makes e.g. returning an i64 between relocatable modules work.

In wasm we don't care about relocatable mode for now, although I do intend to prototype it in the asm2wasm mode soon. So the simplest thing here might be to just add a code path for `WASM_BACKEND` here.